radicle-node/src/signals.rs

```diff
@@ -0,0 +1,55 @@
+use std::io;
+use std::sync::Mutex;
+
+use crossbeam_channel as chan;
+
+/// Signal notifications are sent via this channel.
+static NOTIFY: Mutex<Option<chan::Sender<()>>> = Mutex::new(None);
+
+/// Install global signal handlers for `SIGTERM` and `SIGINT`.
+pub fn install(notify: chan::Sender<()>) -> io::Result<()> {
+    if let Ok(mut channel) = NOTIFY.try_lock() {
+        if channel.is_some() {
+            return Err(io::Error::new(
+                io::ErrorKind::AlreadyExists,
+                "signal handler is already installed",
+            ));
+        }
+        *channel = Some(notify);
+
+        unsafe { _install() }?;
+    } else {
+        return Err(io::Error::new(
+            io::ErrorKind::WouldBlock,
+            "unable to install signal handler",
+        ));
+    }
```

Comment:
What about lifting the nested if statement using the new let statements?

i.e.
```
let Ok(mut channel) = ... else {
  return Err(..)
} 
```

Reply:
Tbh I don't think it really makes a difference here, the function is small..